src/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -438,15 +443,19 @@ private static int FromWin32Value(int value)
         }
 #endif
 
-#if FEATURE_SYSTEM_EVENTS
-        private static void OnUserPreferenceChanging(object sender, UserPreferenceChangingEventArgs e)
+        private static void OnUserPreferenceChanging(object sender, EventArgs args)
         {
-            if (e.Category == UserPreferenceCategory.Color && s_colorTable != null)
+            PropertyInfo categoryProperty = args.GetType()?.GetProperty("Category", BindingFlags.Instance | BindingFlags.Public);
```

Comment:
I think all this light-up code using reflection is worth a comment on why we're doing it this way.